### ISSUE TITLE
false positive on PLC2801 "unnecessary dunder... for __setattr__"

### ISSUE THREAD

**Issue by dimbleby** [astral-sh/ruff#9584]
ruff 0.1.13
```python
from dataclasses import dataclass, field


@dataclass(frozen=True)
class Foo:
    a: int
    b: int
    c: int = field(init=False)

    def __post_init__(self) -> None:
        object.__setattr__(self, "c", self.a + self.b)
```

the `__setattr__` trick is necessary because the dataclass is frozen, but: 
```
PLC2801 Unnecessary dunder call to `__setattr__`. Mutate attribute directly or use setattr built-in function.
```

via https://github.com/astral-sh/ruff/pull/9166 presumably

looks like y'all went round this once already at https://github.com/astral-sh/ruff/pull/4752...

**Comment by trag1c** [comment]
IMO all instances of `object` dunder calls should be allowed considering they're so common (would likely prevent a third round of this issue :)).

**Comment by charliermarsh** [comment]
I believe this is fixed on main, and in v0.1.14 https://play.ruff.rs/73935842-d2c6-4b42-b5de-545c9a5be633.
